### PR TITLE
auto sync 2022-04-20-17-35

### PR DESCRIPTION
**Pull request by davidvossel**
auto sync upstream 2022-04-20-17-35

This PR also has an update to the `hack/sync-upstream.sh` script to use a `git merge` pattern instead of `git rebase`. This gives us better control over how conflicts are handled.

### CODE REVIEW

**Comment by davidvossel** [comment]
@nunnatsa take a look at the new sync logic when you get a chance.

**Comment by davidvossel** [comment]
the e2e tests will fail due to the transition to ginkgo v2

File: hack/sync-upstream.sh [hunk 1]
```diff
@@ -1,95 +1,100 @@
...
+TODAY=$(date -u +%Y-%m-%d-%H-%M)
+NEW_BRANCH_NAME="auto_sync_upstream_${TODAY}"
+
+function verify_merge() {
+	echo "Verifying merge only has expected differences with upstream branch"
+
+	git checkout  $NEW_BRANCH_NAME
+
+	DS_FILES=(OWNERS hack/sync-upstream.sh)
+	diff=$(git diff upstream/main --stat | awk '{ print $1 }')
```
Comment (nunnatsa) [inline hack/sync-upstream.sh:34]: low priority:

```suggestion
	diff=$(git diff upstream/main --name-only)
```

**Review by nunnatsa — APPROVED** [review]

**Pull request merged**